preferences/models.py

```diff
@@ -1,3 +1,41 @@
 from django.db import models
+from django.contrib.auth.models import User
+from users.models import AppUser
 
-# Create your models here.
+from django.utils.translation import gettext_lazy as _
+
+# Model for the database representation of a teacher Preferences record.
+class Preferences(models.Model):
+
+    class SabbaticalLength(models.TextChoices):
+        HALF_LENGTH = 'HALF', _('Half Length')
+        FULL_LENGTH = 'FULL', _('Full Length')
+        NONE = 'NONE', _('None')
+
+    professor = models.OneToOneField(
+        AppUser,
+        on_delete=models.CASCADE,
+        to_field='user',
+        primary_key=True
+    )
+    is_submitted = models.BooleanField(default=False)
+    unavailable_sem1 = models.CharField(max_length=7, blank=True, default='')   #'YYYY-MM'
+    unavailable_sem2 = models.CharField(max_length=7, blank=True, default='')   #'YYYY-MM'
+    num_relief_courses = models.PositiveIntegerField(default=0)
+    taking_sabbatical = models.BooleanField(default=False)
+    sabbatical_length = models.CharField(
+        max_length=4,
+        choices=SabbaticalLength.choices,
+        default=SabbaticalLength.NONE
+    )
+    sabbatical_start_month = models.PositiveIntegerField(default=0)
+    preferred_hours = models.TextField(blank=True)        #holds stringified JSON object
```

Comment:
Might be easier to use built-in json field so you don't have to stringify value. Ex. thttps://docs.djangoproject.com/en/4.0/topics/db/queries/#querying-jsonfield

Reply:
models.JSONField is being used now and seems to work as expected. I prefer this pattern for our model